### DISCUSSION TITLE
Convert PDF into JPEG image without affecting text present in it.

### DISCUSSION THREAD

**Discussion by sbind29893** [ImageMagick/ImageMagick#2946] [Help]
Hi Everyone,

I need help in converting pdf into jpeg image without affecting handwritten text present in it.
Currently, I am using below command but somewhere it is blurring my text in the image.

convert -density 150 input.pdf -quality 100 -deskew 40% output.jpg

Thanks in advance.

**Comment by fmw42** [comment]
try

`convert -density 600 input.pdf -resize 25% -deskew 40% -quality 100 output.jpg`

Note 600 dpi is 4x150 dpi, so then I resize by the inverse, 1/4 = 25%. If you want larger results, resize 50%

Be sure you have a current version of Ghostscript, since ImageMagick uses GS to read PDF files.

If the does not work, then post your input pdf, so we can examine it.

**Reply by sbind29893** [reply]
Hi @fmw42,
thanks for the reply.

I want to convert below kind files into a single PDF and in the further process, I want to split it into a single image and achieve an exact jpg without gridlines and text affected image and with maximum image quality as it was earlier in the original image.

Currently, my image gets a bit blurred when I try to convert my pdf to jpg.

![sample](https://user-images.githubusercontent.com/33458318/101543180-e51c9f00-39c9-11eb-97cd-0b3a55834947.jpg)

**Reply by fmw42** [reply]
@sbind29893 wrote: `Currently, my image gets a bit blurred when I try to convert my pdf to jpg.`

What was your command to convert the PDF to jpg? What is your IM version and platform? What version of Ghostscript are you using with ImageMagick.  Suggested approach

`convert -density 288 img.pdf -resize 25% img.jpg`

if that is not good enough quality, then change 25% to 50% and get twice the image dimensions.

If using IM 7, then change convert to magick.

If the above does not work, post your PDF file.

**Reply by sbind29893** [reply]
@fmw42 

gs 9.25
ImageMagick 6.9.10

ORIGNAL PDF :
[sample.pdf](https://github.com/ImageMagick/ImageMagick/files/5662126/sample.pdf)

convert -density 288 sample.pdf -resize 25% img.jpg
RESULTS :
![img_with_25](https://user-images.githubusercontent.com/33458318/101545303-164a9e80-39cd-11eb-9baf-8841ad1806d6.jpg)

convert -density 288 sample.pdf -resize 50% img_with_50.jpg
RESULTS :
![img_with_50](https://user-images.githubusercontent.com/33458318/101545354-2498ba80-39cd-11eb-90cd-e3cd540393ad.jpg)

**Reply by fmw42** [reply]
Upgrade GS to 9.5x. Your version is old and I believe had some issues. This seems to work well for me.

`convert -density 576 sample.pdf -resize 25% -unsharp 0x1 -quality 92 img.jpg`


![img](https://user-images.githubusercontent.com/13458218/101546327-7ace1500-395d-11eb-8cec-18a78ea8f56d.jpg)

**Reply by sbind29893** [reply]
@fmw42 
Thanks for the suggestion, after upgrading it to 9.5. I am getting good results the same as you.
Can I use the below command with above give the command by you, to remove gridlines and enhance image in a single go?

convert img.jpg -background white -deskew 40% -write mpr:img \
\( mpr:img -morphology close rectangle:30x1 -negate \) \
\( mpr:img -morphology close rectangle:1x30 -negate \) \
-evaluate-sequence add \
result.jpg 

![result](https://user-images.githubusercontent.com/33458318/101549165-2a919a00-39d3-11eb-827e-aeeaf7cca4a2.jpg)

**Comment by snibgo** [comment]
Your sample.pdf contains one page, which contains one JPEG raster image that occupies the entire page.

Converting this with IM involves resampling the page.

But you can use pdfimages to simply extract the JPEG image, without resampling it or changing it in any way.

**Comment by fmw42** [comment]
Yes, if the deskew works adequately. You might also consider what user snibgo said about pdfimages.
